### ISSUE TITLE
Ability to enable and disable Ufe selection of USD globally

### ISSUE THREAD

**Issue by radon199** [Autodesk/maya-usd#1495]
We are starting to use maya_usd with Maya2020 in production and are facing an issue of selection using our older USD workflows but with Maya_usd and UFE. Without access to 2022 in production we are stuck using our old split-proxy workflow, at least for the time being, and that presents issues of selection for us.

The split proxy workflow for us involves using proxy shapes and Maya groups to define the topology of an assembly, and then converting that into a USD layer on export. We had custom changes to the Pixar plugin to enable per prim selection and manipulation, but only when a custom UI was open, otherwise selection was based on the proxy shape.

With maya_usd we are trying to use the same workflow, at least until we can implement a more single stage based workflow, but with the selection modes included as of now in maya_usd we aren't able to make it work exactly as it did before. We want the transformation edits of the whole proxy to go on the proxy shapes's transform, but using the SelectionKind optionVar the highest we can ever get the edit is onto a root prim inside the proxy, using a kind token.

We would like the ability to disable Ufe selection and fall back to selecting the proxy shape, with any of the available plugins. This would allow us to use the per proxy selection we are used to, but also still allow us to disable that for per prim ( or per kind ) selection when artists want that control via Ufe.

Doing a quick prototype it is possible to get the proxy selection behavior by returning false from getInstanceSelectionPath here : https://github.com/Autodesk/maya-usd/blob/06ffd434fc02e01c42e700c2f5c6e8746c0f927f/lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp#L841

While a single proxy workflow is ideal, I still think having a option to support per proxy selection globally is something that might be useful in other cases.

**Comment by kxl-adsk** [comment]
Not sure if you had a chance to dig deeper, but we allow subclasses of base proxy shape to decide if they wish to opt-in for UFE selection or not. This is leveraged by Pixar proxy shape and controls what we return from `ifUfeSelectionEnabled` - check https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/nodes/proxyShapeBase.h#L270

We haven't designed yet any workflow that would require a switch between shape level selection and prim selection, so it's hard to propose a change that we would consider accepting. If you are already building your own version of MayaUSD, you could change the implementation of `ifUfeSelectionEnabled` to include optionVar test. This way from your custom UI you could control the behavior globally.

**Comment by radon199** [comment]
Yes, we have looked into it, and more then likely we will be making the implementation internally.

I still think there would be benefits to making the above change when using Usd Proxy as a object within a deeper maya hierarchy, and where someone might want to separate Maya selection vs UFE section, even in 2022+.

I'm sure lots of different studios found ways around the limitations of the Pixar plugin as it was included in the USD repo. For us it was a multi proxy workflow. Unfortunately the single shape workflow isn't really possible for us without UFE2, unless we go and implement a lot of things that have already been implemented with UFE2 in 2020, and seeing as we just moved to 2020 in production ( from 2018 ) I doubt we will be moving to 2022 in the near future, and so we are faced with the above issue.

If the answer is "we want to focus on 2022+ single stage workflows" then we can work with that, it just means going back to a point where we need to have a number of internal fixes applied on top of the opensource repo which is something we were hoping to avoid going forward, but might just be inevitable due to our dependency on 2020.
